src/profile-logic/profile-data.js

```diff
@@ -417,86 +417,97 @@ export function collapsePlatformStackFrames(thread: Thread): Thread {
-        }
+    const resourceIndex = funcTable.resource[func];
+    const resourceNameIndex = resourceTable.name[resourceIndex];
+    if (resourceNameIndex !== undefined) {
+      const resourceNameString = stringTable.getString(resourceNameIndex);
+      if (resourceNameString.toLowerCase().includes(lowercaseSearchString)) {
+        return true;
       }
+    }
 
-      const resourceIndex = funcTable.resource[func];
-      const resourceNameIndex = resourceTable.name[resourceIndex];
-      if (resourceNameIndex !== undefined) {
-        const resourceNameString = stringTable.getString(resourceNameIndex);
-        if (resourceNameString.toLowerCase().includes(lowercaseSearchString)) {
-          return true;
-        }
-      }
+    return false;
+  }
 
-      return false;
+  const funcMatchesFilterCache = new Map();
+  function funcMatchesFilter(func) {
+    let result = funcMatchesFilterCache.get(func);
+    if (result === undefined) {
+      result = computeFuncMatchesFilter(func);
+      funcMatchesFilterCache.set(func, result);
     }
+    return result;
+  }
 
-    const funcMatchesFilterCache = new Map();
-    function funcMatchesFilter(func) {
-      let result = funcMatchesFilterCache.get(func);
-      if (result === undefined) {
-        result = computeFuncMatchesFilter(func);
-        funcMatchesFilterCache.set(func, result);
-      }
-      return result;
+  const stackMatchesFilterCache = new Map();
+  function stackMatchesFilter(stackIndex) {
+    if (stackIndex === null) {
+      return false;
     }
-
-    const stackMatchesFilterCache = new Map();
-    function stackMatchesFilter(stackIndex) {
-      if (stackIndex === null) {
-        return false;
-      }
-      let result = stackMatchesFilterCache.get(stackIndex);
-      if (result === undefined) {
-        const prefix = stackTable.prefix[stackIndex];
-        if (stackMatchesFilter(prefix)) {
-          result = true;
-        } else {
-          const frame = stackTable.frame[stackIndex];
-          const func = frameTable.func[frame];
-          result = funcMatchesFilter(func);
-        }
-        stackMatchesFilterCache.set(stackIndex, result);
+    let result = stackMatchesFilterCache.get(stackIndex);
+    if (result === undefined) {
+      const prefix = stackTable.prefix[stackIndex];
+      if (stackMatchesFilter(prefix)) {
+        result = true;
+      } else {
+        const frame = stackTable.frame[stackIndex];
+        const func = frameTable.func[frame];
+        result = funcMatchesFilter(func);
       }
-      return result;
+      stackMatchesFilterCache.set(stackIndex, result);
     }
+    return result;
+  }
 
-    return Object.assign({}, thread, {
-      samples: Object.assign({}, samples, {
-        stack: samples.stack.map(s => (stackMatchesFilter(s) ? s : null)),
-      }),
-    });
+  return Object.assign({}, thread, {
+    samples: Object.assign({}, samples, {
+      stack: samples.stack.map(s => (stackMatchesFilter(s) ? s : null)),
+    }),
```

Comment:
Actually nothing changed in this function, only the indentation and the removal of the `timeCode` call, moved to the function `filterThreadToSearchStrings` above.